core/src/test/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunnerTest.java

```diff
@@ -319,6 +321,62 @@ void testDecoratePillarWithClouderaManagerRepoWithNoJsonLicense() throws IOExcep
+        ReflectionTestUtils.setField(underTest, "cmHeartbeatInterval", "testString");
+        ReflectionTestUtils.setField(underTest, "cmMissedHeartbeatInterval", "testString");
+        Set<String> serviceLocations = new HashSet<String>();
+        ClusterView cluster = mock(ClusterView.class);
+        when(stack.getCluster()).thenReturn(cluster);
+        ServiceLocationMap serviceLocationMap = mock(ServiceLocationMap.class);
+        when(clusterApiConnectors.getConnector(cluster)).thenReturn(clusterPreCreationApi);
+        when(clusterPreCreationApi.getServiceLocations()).thenReturn(serviceLocationMap);
+        when(serviceLocationMap.getAllVolumePath()).thenReturn(serviceLocations);
+        when(gatewayConfig.getPrivateAddress()).thenReturn("privateAddress");
+        Map<String, SaltPillarProperties> clouderaManagerSettings = underTest.createPillarWithClouderaManagerSettings(clouderaManagerRepo, stack, gatewayConfig);
+        Object cloudProviderSetupSupported = extractCloudPlatformSupported(clouderaManagerSettings);
+        assertEquals(cloudProviderSetupSupported, Boolean.TRUE);
+    }
+
+    @Test
+    void testCreateCloudManagerSettingsWhenCloudProviderTypeSupportedForYARNCloud() {
+        ClouderaManagerRepo clouderaManagerRepo = new ClouderaManagerRepo();
+        GatewayConfig gatewayConfig = mock(GatewayConfig.class);
+        clouderaManagerRepo.setVersion("7.6.2");
+        when(stack.getCloudPlatform()).thenReturn(CloudPlatform.YARN.name());
+        ClusterPreCreationApi clusterPreCreationApi = mock(ClusterPreCreationApi.class);
+        when(stack.getStack().getResourceCrn()).thenReturn(TEST_CLUSTER_CRN);
+        when(entitlementService.cmAutoBundleCollectionDisabled(anyString())).thenReturn(Boolean.TRUE);
+        ReflectionTestUtils.setField(underTest, "cmHeartbeatInterval", "testString");
+        ReflectionTestUtils.setField(underTest, "cmMissedHeartbeatInterval", "testString");
+        Set<String> serviceLocations = new HashSet<String>();
+        ClusterView cluster = mock(ClusterView.class);
+        when(stack.getCluster()).thenReturn(cluster);
+        ServiceLocationMap serviceLocationMap = mock(ServiceLocationMap.class);
+        when(clusterApiConnectors.getConnector(cluster)).thenReturn(clusterPreCreationApi);
+        when(clusterPreCreationApi.getServiceLocations()).thenReturn(serviceLocationMap);
+        when(serviceLocationMap.getAllVolumePath()).thenReturn(serviceLocations);
+        when(gatewayConfig.getPrivateAddress()).thenReturn("privateAddress");
+        Map<String, SaltPillarProperties> clouderaManagerSettings = underTest.createPillarWithClouderaManagerSettings(clouderaManagerRepo, stack, gatewayConfig);
+        Object cloudProviderSetupSupported = extractCloudPlatformSupported(clouderaManagerSettings);
+        assertEquals(cloudProviderSetupSupported, Boolean.FALSE);
+    }
+
+    private Object extractCloudPlatformSupported(Map<String, SaltPillarProperties>clouderaManagerSettings) {
```

Comment:
why did you change to Object? It was a Boolean before.
It should be a primitive `boolean`